src/Compilers/CSharp/Test/Syntax/Parsing/DeclarationParsingTests.cs

```diff
@@ -7266,6 +7266,52 @@ public void TestOptParamMethodDeclarationWithNullValidation()
             }
         }
 
+        [Fact]
+        public void TestOptParamMethodDeclarationWithNullValidationNoSpaces()
+        {
+            UsingStatement(@"void M(string name!=null) { }", expectedErrors: new DiagnosticDescription[]
```

Comment:
there needs ot be a lot more tests here.  for example `name !=null` `name<new-line>!=null` `name!= null` `name!=<newline>null`.  As well as tests with comments and whatnot around the token.  We need to ensure that tree invariants are preserved.

Reply:
In progress

Reply:
thanks!